pkg/server/server.go

```diff
@@ -249,6 +255,36 @@ func (s *Server) Start(ctx context.Context) error {
 	}
 	logrus.WithFields(logrus.Fields{"id": s.app.ID(), "address": s.app.Address()}).Print("Started dqlite")
 
+	go func() {
+		for {
+			select {
+			case <-storageWatchTicker.C:
+				var stat unix.Statfs_t
+				unix.Statfs(s.storageDir, &stat)
+				avail := stat.Bavail * uint64(stat.Bsize)
+				// If available space is less than 10MB reject to start.
+				if avail < disk_threshold {
+					// Create a separate context with 30 seconds to cleanup
+					stopCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+					defer cancel()
+
+					logrus.Debug("Handing over dqlite leadership")
+					if err := s.app.Handover(stopCtx); err != nil {
```

Comment:
What happens if you are not a leader?

Reply:
Go-dqlite handles this in the background, only voter and leadership rights are handed over. Please see [app.go#L311-L316](https://github.com/canonical/go-dqlite/blob/9e965d50ce079bea63e8c76b18c07f3ca91fc2e4/app/app.go#L311-L316)